### PR TITLE
NIAD-45: Set timeout and retries for LDAP connection

### PR DESCRIPTION
**Pull request by grzegorz-robak**

### CODE REVIEW

File: mhs/spineroutelookup/lookup/sds_connection_factory.py [hunk 1]
```diff
@@ -5,12 +5,15 @@
 
 import definitions
 
+_MAX_RETRIES = 3
+_CONNECTION_TIMEOUT = 5
```
Comment (bartek-sarul) [inline mhs/spineroutelookup/lookup/sds_connection_factory.py]: in such case it's worth adding a comment saying the unit of time span used

**Review by bartek-sarul — CHANGES_REQUESTED** [review]

File: mhs/spineroutelookup/lookup/sds_connection_factory.py (hunk 1 above)
Comment (grzegorz-robak) [reply mhs/spineroutelookup/lookup/sds_connection_factory.py]: Good idea!

**Review by jaroslaw-wojciechowski — DISMISSED** [review]

**Review by jaroslaw-wojciechowski — DISMISSED** [review]

File: mhs/spineroutelookup/lookup/sds_connection_factory.py [hunk 2]
```diff
@@ -5,12 +5,15 @@
 
 import definitions
 
+_MAX_RETRIES = 3
```
Comment (bartek-sarul) [inline mhs/spineroutelookup/lookup/sds_connection_factory.py]: I was looking at code here and a common approach is to load values from environment variables but override with default one if env var is not provided
For example
https://github.com/nhsconnect/integration-adaptors/pull/153/files#diff-8a04b7e2f084c2f292ffdc46a46ccc61R88

**Review by bartek-sarul — CHANGES_REQUESTED** [review]

File: mhs/spineroutelookup/lookup/sds_connection_factory.py (hunk 2 above)
Comment (grzegorz-robak) [reply mhs/spineroutelookup/lookup/sds_connection_factory.py]: Done. Added some additional logging also.

**Review by bartek-sarul — DISMISSED** [review]

**Pull request merged**